pinot-common/src/main/java/org/apache/pinot/common/function/FunctionRegistry.java

```diff
@@ -91,29 +86,28 @@ private FunctionRegistry() {
   public static void init() {
   }
 
+  public static void registerFunction(Method method) {
+    registerFunction(method.getName(), method);
+  }
+
   /**
    * Registers a method with the name of the method.
    */
-  public static void registerFunction(Method method, boolean nullableParameters) {
-    registerFunction(method.getName(), method, nullableParameters);
-
-    // Calcite ScalarFunctionImpl doesn't allow customized named functions. TODO: fix me.
+  public static void registerFunction(String name, Method method) {
     if (method.getAnnotation(Deprecated.class) == null) {
```

Comment:
(MAJOR) This will cause deprecated method not registered, which is backward incompatible